### PR TITLE
 Fold jingo_minify into our code

### LINKED ISSUE #8532: Fold jingo_minify into addons-server
In preparation and as part of our upgrade to Django 1.11 we should fold jingo_minify into addons-server to eventually get rid of it and get more control over it.

### PR DESCRIPTION
**Pull request by EnTeQuAk**
This will simplify the upgrade to Django 1.11 and allow us to keep this
under control, upstream isn't properly maintained.

* Copies jingo-minify into our code
* Removed all parts we aren't using (yui, git integration, `{{ inline_css }}` template tag etc)
* Try to clean up how subprocess calls are being made
* Cleaned up tests

This doesn't change anything regarding how jingo-minify works. It's still using good old `cat` and other ugly things but the code now looks nicer.

See https://github.com/mozilla/addons-server/pull/8694/commits/742297fef9d75ade43ce0b416d9b645d59a71e61 for more specific details about the cleanup from the original jingo-minify dump.

Fixes #8532

### CODE REVIEW

**Comment by diox** [comment]
> Removed all parts we aren't using (uglifyjs, git integration etc)

Wait we're using uglifyjs, you removed yui right ? :)

**Comment by EnTeQuAk** [comment]
> Wait we're using uglifyjs, you removed yui right ? :)

Haha, yeah I did :rofl:

File: conftest.py [hunk 1]
```diff
@@ -60,21 +60,6 @@ def mock_basket(settings):
...
-    When testing, we don't want nor need the bundled/minified css files, so
-    pretend that all the css files are external.
-
-    Mocking this will prevent amo.templatetags.jinja_helpers.inline_css to
-    believe it should bundle the css.
-
-    """
-    from olympia.amo.templatetags import jinja_helpers
-    monkeypatch.setattr(jinja_helpers, 'is_external', lambda css: True)
-
```
Comment (EnTeQuAk) [inline conftest.py]: We're not using `{{ inline_css }}` at all so I removed it :man_shrugging:

**Comment by EnTeQuAk** [comment]
See https://github.com/mozilla/addons-server/pull/8694/commits/742297fef9d75ade43ce0b416d9b645d59a71e61 for more specific details about the cleanup from the original jingo-minify dump.

File: src/olympia/amo/management/commands/compress_assets.py [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+            os.path.join(settings.ROOT, 'build.py'))
+
+        with open(build_id_file, 'w') as f:
+            f.write('BUILD_ID_CSS = "%s"\n' % self.build_id)
+            f.write('BUILD_ID_JS = "%s"\n' % self.build_id)
+            f.write('BUILD_ID_IMG = "%s"\n' % self.build_id)
+            f.write('BUNDLE_HASHES = %s\n' % self.bundle_hashes)
+
+    def handle(self, **options):
+        self.build_id = self.generate_build_id()
```
Comment (eviljeff) [inline src/olympia/amo/management/commands/compress_assets.py]: nit-ish: We generate `self.build_id` with `generate_build_id` here but only use it in `update_hashes` where we set it again.  Seems like we only need to call `generate_build_id` in `update_hashes`.

File: src/olympia/lib/settings_base.py [hunk 3]
```diff
@@ -1719,11 +1716,11 @@ def get_redis_settings(uri):
 # Static
 STATIC_ROOT = path('site-static')
 STATIC_URL = '/static/'
-JINGO_MINIFY_ROOT = path('static')
```
Comment (eviljeff) [inline src/olympia/lib/settings_base.py]: Just checking, we can still edit css & js files in `/static/` and use them locally, right?

File: src/olympia/amo/management/commands/compress_assets.py (hunk 2 above)
Comment (EnTeQuAk) [reply src/olympia/amo/management/commands/compress_assets.py]: Indeed, that was left from the refactor of the git integration. Nice find!

File: src/olympia/lib/settings_base.py (hunk 3 above)
Comment (EnTeQuAk) [reply src/olympia/lib/settings_base.py]: Yes, we do. I tested this and it seemed to work nicely, tested both `site-static` and `static` still works from my perspective.

File: src/olympia/amo/management/commands/compress_assets.py [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+    """Run a command and correctly poll the output and write that to stdout"""
+    process = subprocess.Popen(command, stdout=subprocess.PIPE, shell=True)
+    while True:
+        output = process.stdout.readline()
+        if output == '' and process.poll() is not None:
+            break
+        if output:
+            print(output.strip())
+    rc = process.poll()
+    return rc
```
[the remark this answers is not in the document]
Comment (EnTeQuAk) [reply src/olympia/amo/management/commands/compress_assets.py]: Also fixed (with the other stuff) in 7ab68884af

File: src/olympia/amo/management/commands/compress_assets.py [hunk 5]
```diff
@@ -65,8 +64,6 @@ def update_hashes(self):
             f.write('BUNDLE_HASHES = %s\n' % self.bundle_hashes)
 
     def handle(self, **options):
-        self.build_id = self.generate_build_id()
```
Comment (eviljeff) [inline src/olympia/amo/management/commands/compress_assets.py]: 😨 I don't think you meant to do just that - `self.build_id = '%s-%s' % (self.build_id, hex(int(time.time()))[2:])` needs to be changed too.

Comment (EnTeQuAk) [reply src/olympia/amo/management/commands/compress_assets.py]: > I don't think you meant to do just that - self.build_id = '%s-%s' % (self.build_id, hex(int(time.time()))[2:]) needs to be changed too.

Eh, I certainly didn't. Fix committed in abdc808

**Pull request merged**